src/components/Navigation/NavigationTabBar/index.tsx

```diff
@@ -116,8 +116,24 @@ function NavigationTabBar({selectedTab, isTooltipAllowed = false, isTopLevelBar
         }
 
         hideInboxTooltip();
+        if (shouldUseNarrowLayout) {
+            Navigation.navigate(ROUTES.HOME);
+            return;
+        }
+
+        const rootState = navigationRef.getRootState() as State<RootNavigatorParamList>;
+        const lastReportNavigator = rootState.routes.findLast((route) => route.name === NAVIGATORS.REPORTS_SPLIT_NAVIGATOR);
+        const lastReportNavigatorState = lastReportNavigator && lastReportNavigator.key ? getPreservedNavigatorState(lastReportNavigator?.key) : undefined;
+        const lastReportRoute = lastReportNavigatorState?.routes.findLast((route) => route.name === SCREENS.REPORT);
+
+        if (lastReportRoute) {
+            const {reportID} = lastReportRoute.params as ReportsSplitNavigatorParamList[typeof SCREENS.REPORT];
+            Navigation.navigate(ROUTES.REPORT_WITH_ID.getRoute(reportID));
+            return;
+        }
```

Comment:
@adamgrzybowski @WojtekBoman is this the cleanest way to write this logic? @bernhardoj Can you please add plain English comments here too? We should explain what this is doing and why. Thank you

Reply:
Seems like preloading is quite close so I will hold this PR